### PR TITLE
Tests: balance forest

### PR DESCRIPTION
**Pull request by Niklas997**
**_Describe your changes here:_**
Added some tests regarding the balance functionality.

This covers a test for expected behaviour on a simple un-balanced forest as well as some tests of the boolean _t8_forest_is_balanced_ function and a consistency test for an already balanced forest.

Some of the balance functionality was already called within the test _t8_gtest_forest_commit_ which was not added to the testsuite beforehand.

**_All these boxes must be checked by the reviewers before merging the pull request:_**

As a reviewer please read through all the code lines and make sure that the code is fully understood, bug free, well-documented and well-structured.

#### General
- [ ] The reviewer executed the new code features at least once and checked the results manually

- [ ] The code follows the [t8code coding guidelines](https://github.com/holke/t8code/wiki/Coding-Guideline)
- [ ] New source/header files are properly added to the Makefiles
- [ ] The code is well documented
- [ ] All function declarations, structs/classes and their members have a proper doxygen documentation
- [ ] All new algorithms and data structures are sufficiently optimal in terms of memory and runtime (If this should be merged, but there is still potential for optimization, create a new issue)

#### Tests
- [ ] The code is covered in an existing or new test case using Google Test

#### Github action

- [ ] The code compiles without warning in debugging and release mode, with and without MPI (this should be executed automatically in a github action)
- [ ] All tests pass (in various configurations, this should be executed automatically in a github action)

  If the Pull request introduces code that is not covered by the github action (for example coupling with a new library):
  - [ ] Should this use case be added to the github action?
  - [ ] If not, does the specific use case compile and all tests pass (check manually)

#### Scripts and Wiki

- [ ] If a new directory with source-files is added, it must be covered by the `script/find_all_source_files.scp` to check the indentation of these files.
- [ ] If this PR introduces a new feature, it must be covered in an example/tutorial and a Wiki article.

#### Licence

- [ ] The author added a BSD statement to `doc/` (or already has one)

### CODE REVIEW

File: test/t8_forest/t8_gtest_balance.cxx [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+  {
+    return 1;
+  } else
+  {
+    return 0;
+  }
+}
+
+/**
+ * \brief This function generates a simple forest which is not balanced.
```
Comment (jmark) [inline test/t8_forest/t8_gtest_balance.cxx:99]: Nice documentation! :)

File: test/t8_forest/t8_gtest_balance.cxx [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+  t8_forest_ref(balanced_forest);
+
+  flag_no_repartition = 1;
+  t8_forest_t already_balanced_forest;
+  t8_forest_init(&already_balanced_forest);
+  t8_forest_set_balance(already_balanced_forest, balanced_forest, flag_no_repartition);
+  t8_forest_commit(already_balanced_forest);
+
+  EXPECT_TRUE(t8_forest_is_equal(balanced_forest, already_balanced_forest) == 1);
```
Comment (jmark) [inline test/t8_forest/t8_gtest_balance.cxx:241]:
```suggestion
  EXPECT_EQ(t8_forest_is_equal(balanced_forest, already_balanced_forest), 1);
```

**Review by jmark — CHANGES_REQUESTED** [review]

File: test/Makefile.am [hunk 3]
```diff
@@ -58,6 +58,8 @@ t8code_googletest_programs = \
   test/t8_forest/t8_gtest_ghost_exchange \
   test/t8_forest/t8_gtest_ghost_delete \
   test/t8_forest/t8_gtest_ghost_and_owner \
+  test/t8_forest/t8_gtest_forest_commit \
```
[the remark this answers is not in the document]
Comment (Niklas997) [reply test/Makefile.am:61]: The test was already implemented. However, it was not added to the makefile beforehand. I just made the addition to the makefile, this could have been a PR on it's own, but it is somehow linked in terms of the test coverage of the balance functionality.

File: test/t8_forest/t8_gtest_balance.cxx [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+  }
+  t8_eclass_t ieclass;
+};
+
+/**
+ * \brief This test confirms that the function 't8_forest_is_balanced' recognizes uniform forests as balanced.
+ */
+TEST_P(gtest_balance, confirm_is_balanced_check_for_uniform_forests)
+{
+  const int level = 3;
```
[the remark this answers is not in the document]
Comment (Niklas997) [reply test/t8_forest/t8_gtest_balance.cxx:60]: I thought about that too that we could test more levels or even parametrize the cmesh. But I was not sure, if the test becomes too large and especially if there is a "testing-benefit" when we vary the uniform refinement level. What do you think? Should we add more cases to the test?

[the remark this answers is not in the document]
Comment (jmark) [reply test/t8_forest/t8_gtest_balance.cxx:60]: I would at least test levels `0` to `3`. Sometimes level `0` breaks the logic. Additionally, I would also test a cmesh with periodic boundaries.

**Review by jmark — APPROVED** [review]
